src/backend/tests/projects.test.ts

```diff
@@ -666,4 +667,46 @@ describe('Projects', () => {
       expect(prisma.material_Type.create).toBeCalledTimes(1);
     });
   });
+
+  describe('assigning material assemblies', () => {
+    test('assignment fails because of permissions', async () => {
+      vi.spyOn(prisma.material, 'findUnique').mockResolvedValue(prismaMaterial1);
+      vi.spyOn(prisma.assembly, 'findUnique').mockResolvedValue(prismaAssembly1);
+      vi.spyOn(prisma.project, 'findFirst').mockResolvedValue({ teams: [prismaTeam1] });
+
+      await expect(ProjectsService.assignMaterialAssembly(theVisitor, 'mid', 'aid')).rejects.toThrow(
+        new AccessDeniedException('Only leadership or above can create a material type')
+      );
+    });
```

Comment:
This test is still failing

Reply:
The reason that we're getting the error from Github Actions is b/c we want to set `prisma Project` type value inside mockResolvedVAlue for prisma.project's findFirst spyOn. If we want to set different teams prop in the project variable, we can set {...prismaProject1, teams: [prismaTeam1]} as the mockResolvedValue for ex.